src/H5Dint.c

```diff
@@ -2729,9 +2729,8 @@ H5D__vlen_get_buf_size(H5D_t *dset, hid_t type_id, hid_t space_id, hsize_t *size
     vlen_bufsize.common.size = 0;
 
     /* Call H5S_select_iterate with args, etc. */
-    dset_op.op_type          = H5S_SEL_ITER_OP_APP;
-    dset_op.u.app_op.op      = H5D__vlen_get_buf_size_cb;
-    dset_op.u.app_op.type_id = type_id;
+    dset_op.op_type  = H5S_SEL_ITER_OP_LIB;
+    dset_op.u.lib_op = H5D__vlen_get_buf_size_cb;
```

Comment:
It didn't look like there was a need for the callback here to be an application iter op, as opposed to a library one. If I left this as an application op, the passed in `type_id` would have to be unwrapped for every single element so that the `H5T_t *` can be assigned to `dset_info.mem_type` in the changes above to `H5D__vlen_get_buf_size_cb`. Switching this to a library iter op passes an `H5T_t *` to the callback which is much simpler.

Reply:
Agree, good call